bin/add_frontend_server.rb

```diff
@@ -0,0 +1,93 @@
+require 'aws-sdk'
+require_relative '../deployment'
+
+MAX_WAIT_TIME = 600 #Wait no longer than 10 minutes for instance creation. Typically this takes a couple minutes
+
+ec2client = Aws::EC2::Client.new
+
+instances = ec2client.describe_instances
+
+#Determine distribution of availability zones, pick the one that has the least capacity among frontend instances
+frontend_instances = []
+
+instances.reservations.each do |reservation|
+  frontends_for_reservation = reservation.instances.select{|instance| instance.tags.detect{|tag| tag.key == 'Name' && tag.value.include?('frontend')}}
+  frontend_instances << frontends_for_reservation unless frontends_for_reservation.size == 0
+end
+
+instance_distribution = frontend_instances.each_with_object(Hash.new(0)){|(instance, _), instance_distribution| instance_distribution[instance.placement.availability_zone] += 1}
+
+determined_instance_zone = instance_distribution.select { |_, v| v == instance_distribution.values.min }.keys[0]
```

Comment:
Isn't the instance count used on line [#24](https://github.com/code-dot-org/code-dot-org/pull/4015#discussion-diff-39783016R24) below?